vocabulary-draft/continuous-deployment-pipeline-events.md

```diff
@@ -0,0 +1,26 @@
+# Continuous Deployment Pipelines Events 
+
+__Note:__ This is a work-in-progress draft version and is being worked on by members of the Events SIG. You are very welcome to join the work and the discussions!
+
+These events are related to continous deployment pipelines and their target environments. 
+These events can be emitted by environments to report where software artifacts such as services, binaries, deamons, jobs or embedded software are running. 
+
+The term Service is used to represent a running Artifact. This service can represent a binary that is running, a deamon, an application, a docker container, etc.
+The term Environment represent any platform which has all the means to run a Service. 
+
+- **Environment Created**: an environment has been created and it can be used to deploy Services
+- **Environment Modified**: an environment has been modified, this event advertise the changes made in the environment
+- **Environment Deleted**: an environment has been deleted and cannot longer be used
+- **Service Deployed**: a new instance of the Service has been deployed
+- **Service Upgraded**: an existing instance of a Service has been upgraded to a new version
```

Comment:
Should we have rollback too? Or a single version changed event which could be used for both?
Does changing the configuration count as upgraded?
I think many systems won't allow to distinguish between version change and config change, so perhaps those could be optional attributes in the event body.

Reply:
@afrittoli I think that having rollback makes sense to make it more explicit.. We can definitely deal with going back in version with Upgraded as well, but it will be confusing for some people. I am adding Service Rollback.